Core/include/Acts/TrackFinding/TrackParamsLookupAccumulator.hpp

```diff
@@ -0,0 +1,217 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/EventData/GenericBoundTrackParameters.hpp"
+#include "Acts/EventData/TrackParametersConcept.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+
+#include <concepts>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <stdexcept>
+#include <utility>
+
+namespace {
```

Comment:
Anonymous namespaces don't do a lot in headers. Can you move this in a `detail` namespace here?

Reply:
Moved the concepts into a separate file, so that they are reusable in `TrackParametersJsonConverter`